src/main/java/co/rsk/bitcoinj/script/FastBridgeParser.java

```diff
@@ -0,0 +1,113 @@
+package co.rsk.bitcoinj.script;
+
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.VerificationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class FastBridgeParser extends StandardRedeemScriptParser {
+    private static final Logger logger = LoggerFactory.getLogger(FastBridgeParser.class);
+    protected final byte[] derivationHash;
+
+    public FastBridgeParser(
+        ScriptType scriptType,
+        List<ScriptChunk> redeemScriptChunks,
+        List<ScriptChunk> rawChunks
+    ) {
+        super(
+            scriptType,
+            redeemScriptChunks,
+            rawChunks
+        );
+
+        // the idea of this was to not have to call the extract function and check the type again
+        // so when we call an instance of FastBridgeParser, the multisigType and the redeemScriptChunks are calculated asap
+        // and when we call just the extract function from outside, it does the multisig type verification
+
+        List<ScriptChunk> subChunks = redeemScriptChunks.subList(2, redeemScriptChunks.size());
+        if (isFastBridgeMultiSig(redeemScriptChunks)) {
+            this.multiSigType = MultiSigType.FAST_BRIDGE_MULTISIG;
```

Comment:
The way I see it we should a single MultiSigType for flyover. Do you think that's feasible? So we avoid adding new types for flyover everytime there is a new script type

Reply:
I absolutely agree. Mainly because you can simply remove the flyover values and check for the type as usual.